packages/frontend/core/src/layouts/workspace-layout.tsx

```diff
@@ -121,6 +124,27 @@ export const WorkspaceLayoutInner = ({ children }: PropsWithChildren) => {
     workbench.location$.map(location => basename + location.pathname)
   );
 
+  useEffect(() => {
+    const disposable = AIProvider.slots.requestInsertTemplate.on(
+      ({ template, mode }) => {
+        (async () => {
+          const templateZip = await fetch(template);
```

Comment:
Does this take long? Maybe add a toast?

Reply:
It may be need a toast, but this feature is behind feature flag, so we can try it and see.